### ISSUE TITLE
Waterline doesn't work with io.js (in strict mode)

### ISSUE THREAD

**Issue by teologov** [balderdashy/waterline#1080]
Hi, I tried to run Waterline in the strict mode, using `io.js` v2.3.3

Unfortunately, it fails with this error: 
`.../waterline-express-test/node_modules/waterline/node_modules/anchor/node_modules/gjtk/node_modules/uri-js/src/uri.js:51
URI = (function () {
    ^
ReferenceError: URI is not defined
    at Object.<anonymous> (/Users/teologov/Sites/personal/other/waterline-express-test/node_modules/waterline/node_modules/anchor/node_modules/gjtk/node_modules/uri-js/src/uri.js:51:5)
    at Module._compile (module.js:430:26)
    at Object.Module._extensions..js (module.js:448:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/teologov/Sites/personal/other/waterline-express-test/node_modules/waterline/node_modules/anchor/node_modules/gjtk/GeoJSON.js:2:11)`

`gjtk` package uses outdated `uri-js@1.4.2` dependency, which latest version is `2.1.0`.

**Comment by tjwebb** [comment]
I'm not sure where to begin here. Any experts on io.js? 

io.js releases a new version like every week, with potentially breaking changes with every release. There's really no way waterline can keep up with that kind of release schedule.

> gjtk package uses outdated uri-js@1.4.2 dependency, which latest version is 2.1.0.

So, is the problem actually with `gjtk`?

**Comment by teologov** [comment]
> So, is the problem actually with gjtk?

It seems so. I'll try to update `gjtk`'s `uri-js` dependency to the latest major version. Hopefully, it will resolve this issue.

**Comment by teologov** [comment]
Hi, I've got some news regarding this issue.

What has been already done:
* `gjtk` package is fixed, and successfully published.
* `uri-js` is fixed, and published as well.

What should be accomplished:
* I've created pull request to `sails-disk`: https://github.com/balderdashy/sails-disk/pull/45. It's not merged in master yet.
* Update `anchor`'s (I mean https://github.com/balderdashy/anchor) `gjtk` dependency version to the latest version.
* Finally, fix `waterline-schema` module. I found out, that it's failing as well.

With that said, I've just gathered all fixed modules in the one test project, and waterline works locally without any problem with `io.js v.2.3.3`.

Hopefully, tomorrow I will create all the rest of pull requests.

**Comment by devinivy** [comment]
Awesome– many thanks, @teologov .  I'll be on the lookout for those PRs.

**Comment by teologov** [comment]
Hey guys. I've created all required pull request. 
Actually, I had to create one more pull request to `switchback` module: https://github.com/node-machine/switchback/pull/4.

**Comment by ezsper** [comment]
Same issue with iojs `2.5.0` and `3.0.0` with waterline `0.10.25` and `0.10.26`, I'm using babel `5.4.7` but not for `node_modules`, so it's an iojs incompatibility, could we have this fixed? I don't want to fork this only to make it work and I also want to keep using iojs.

    [piping] error given was: ReferenceError: URI is not defined
        at Object.<anonymous> (./node_modules/waterline/node_modules/anchor/node_modules/gjtk/node_modules/uri-js/src/uri.js:51:4)
        at Module._compile (module.js:430:26)
        at normalLoader (./node_modules/babel-core/lib/babel/api/register/node.js:150:5)
        at Object.require.extensions.(anonymous function) [as .js] (./node_modules/babel-core/lib/babel/api/register/node.js:163:7)
        at Module.load (module.js:355:32)
        at Module._load (module.js:310:12)
        at Function.module._load (./node_modules/piping/lib/launcher.js:32:16)
        at Module.require (module.js:365:17)
        at require (module.js:384:17)
        at Object.<anonymous> (./node_modules/anchor/node_modules/gjtk/GeoJSON.js:2:11)
        at Module._compile (module.js:430:26)
        at Module._extensions..js (module.js:448:10)
        at Object.require.extensions.(anonymous function) [as .js] (./node_modules/babel-core/lib/babel/api/register/node.js:161:7)
        at Module.load (module.js:355:32)
        at Module._load (module.js:310:12)
        at Function.module._load (./node_modules/piping/lib/launcher.js:32:16)

**Comment by ezsper** [comment]
Also getting this issue form nodejs `0.10.40`

    URI = (function () {
        ^
    ReferenceError: URI is not defined
        at Object.<anonymous> (./node_modules/waterline/node_modules/anchor/node_modules/gjtk/node_modules/uri-js/src/uri.js:51:4)
        at Module._compile (module.js:456:26)
        at normalLoader (./node_modules/babel-core/lib/babel/api/register/node.js:150:5)
        at Object.require.extensions.(anonymous function) [as .js] (./node_modules/babel-core/lib/babel/api/register/node.js:163:7)
        at Module.load (module.js:356:32)
        at Function.Module._load (module.js:312:12)
        at Module.require (module.js:364:17)
        at require (module.js:380:17)
        at Object.<anonymous> (./node_modules/waterline/node_modules/anchor/node_modules/gjtk/GeoJSON.js:2:11)
        at Module._compile (module.js:456:26)
        at Module._extensions..js (module.js:474:10)
        at Object.require.extensions.(anonymous function) [as .js] (./node_modules/babel-core/lib/babel/api/register/node.js:161:7)
        at Module.load (module.js:356:32)
        at Function.Module._load (module.js:312:12)
        at Module.require (module.js:364:17)
        at require (module.js:380:17)

**Comment by sailsbot** [comment]
Thanks for posting, @teologov.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
